src/segment_evaluation.rs

```diff
@@ -316,11 +276,17 @@ pub mod tests {
         //  Caused by: Operation 'name' 'is' 'heinz' failed to evaluate.
         //  Caused by: Entity attribute has unexpected type: Number.
         // We are checking here that the parts are present to allow debugging of config by the user:
-        let msg = rule.unwrap_err().to_string();
-        assert!(msg.contains("'a2'"));
-        assert!(msg.contains("'0'"));
-        assert!(msg.contains("'some_segment_id_1'"));
-        assert!(msg.contains("'name' 'is' 'heinz'"));
-        assert!(msg.contains("Entity attribute has unexpected type: Number"));
+
+        let e = rule.unwrap_err();
+        assert!(matches!(e, Error::EntityEvaluationError(_)));
+        let Error::EntityEvaluationError(EntityEvaluationError(
+            SegmentEvaluationError::SegmentEvaluationFailed(ref error),
+        )) = e
+        else {
+            panic!("Error type mismatch!");
+        };
+        assert_eq!(error.segment.name, "");
```

Comment:
Not sure why this is empty (we have a `Segment` object). Is `name` actually empty?

Reply:
I was going with segment ID for debugging always. in the unit-tests the name is empty string. only in the "real" json test data it is set.